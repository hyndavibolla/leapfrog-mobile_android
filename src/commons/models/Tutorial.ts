import { ReactNode } from 'react';
import { StyleProp, ViewStyle } from 'react-native';

export interface Tutorial {
  isParent?: boolean;
  isHidden?: boolean;
  parentId?: number;
  scrollToDown?: number;
}

export interface TutorialElementPosition extends Tutorial {
  x: number;
  y: number;
  children: ReactNode;
  style: StyleProp<ViewStyle>;
}
