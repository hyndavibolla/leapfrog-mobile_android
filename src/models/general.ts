import { Dispatch } from 'react';
import Auth0 from 'react-native-auth0';

import { ApiService } from '_services/ApiService';
import { AppService } from '_services/AppService';
import { CCPAService } from '_services/CCPAService';
import { ICrashlyticsService } from '_services/CrashlyticsService';
import { EventTrackerService } from '_services/EventTrackerService';
import { IHttpService } from '_services/HttpService';
import { NativeHelperService } from '_services/NativeHelperService';
import { Logger } from '_services/Logger';
import { IPerformanceTrackingService } from '_services/PerformanceTrackingService';
import { RaiseService } from '_services/RaiseService';
import { RemoteConfigService } from '_services/RemoteConfigService';
import { ICardLinkState } from '_state_mgmt/cardLink/state';
import { ICoreState } from '_state_mgmt/core/state';
import { IGameState } from '_state_mgmt/game/state';
import { IGiftCardState } from '_state_mgmt/giftCard/state';
import { IMessagesState } from '_state_mgmt/messages/state';
import { IMissionState } from '_state_mgmt/mission/state';
import { IRewardState } from '_state_mgmt/reward/state';
import { ISurveyState } from '_state_mgmt/survey/state';
import { IUserState } from '_state_mgmt/user/state';

export interface IAction<T = string, P = any> {
  type: T;
  payload: P;
}

export type IReducer<S = any, A = any> = (state: S, action: IAction<A>) => S;

export interface Deps {
  apiService: ApiService;
  appService: AppService;
  crashlyticsService: ICrashlyticsService;
  httpService: IHttpService;
  performanceTrackingService: IPerformanceTrackingService;
  raiseService: RaiseService;
  nativeHelperService: NativeHelperService;
  stateSnapshot: { get: () => IGlobalState; set: (state: IGlobalState) => void };
  logger: Logger;
  eventTrackerService: EventTrackerService;
  ccpaService: CCPAService;
  remoteConfigService: RemoteConfigService;
  authService: Auth0;
}
export type IGlobalState = {
  core: ICoreState;
  user: IUserState;
  game: IGameState;
  mission: IMissionState;
  reward: IRewardState;
  survey: ISurveyState;
  messages: IMessagesState;
  cardLink: ICardLinkState;
  giftCard: IGiftCardState;
};

export interface IGlobalContext {
  state: IGlobalState;
  dispatch: Dispatch<IAction<any, any>>;
  deps: Deps;
}

export enum HTTP_METHOD {
  GET = 'GET',
  POST = 'POST',
  PUT = 'PUT',
  PATCH = 'PATCH',
  DELETE = 'DELETE'
}

export interface ITokenSet {
  accessToken: string;
  refreshToken: string;
  accessTokenExpiryTime: string;
  sywToken?: string;
  /**
   * @deprecated deviceId does not belong to ITokenSet anymore.
   */
  deviceId?: string;
  scope?: string;
  expiresIn?: number;
  tokenType?: string;
}

export enum FeatureFlag {
  API_OVERRIDE = 'API_OVERRIDE',
  TEST_ONLY = 'TEST_ONLY',
  ENV_BANNER = 'ENV_BANNER',
  BUTTON_DEBUG = 'BUTTON_DEBUG',
  LOG_LEVEL_INFO = 'LOG_LEVEL_INFO',
  LOG_LEVEL_DEBUG = 'LOG_LEVEL_DEBUG',
  LOG_LEVEL_WARN = 'LOG_LEVEL_WARN',
  LOG_LEVEL_ERROR = 'LOG_LEVEL_ERROR',
  LOG_VIEW = 'LOG_VIEW',
  STREAK_REDEEM_CARD = 'STREAK_REDEEM_CARD',
  SAILTHRU_TRACKING = 'SAILTHRU_TRACKING',
  SURVEY = 'SURVEY',
  PREDICTIVE_MAP_SEARCH = 'PREDICTIVE_MAP_SEARCH',
  MISSIONS_SEARCH = 'MISSIONS_SEARCH',
  SEE_ALL_TRANSACTIONS = 'SEE_ALL_TRANSACTIONS',
  WALLET_YOUR_GC = 'WALLET_YOUR_GC'
}

export type DateLike = Date | string | number;

export type HTMLString = string;

export interface IGameLevelUI {
  updatedAt: DateLike;
  level: number;
}

export interface ICCPASetting {
  allow: boolean;
}

export interface IOnboarding {
  seenOnboarding: boolean;
}

export enum ButtonCreativeType {
  HERO = 'hero',
  CAROUSEL = 'carousel',
  LIST = 'list',
  GRID = 'grid',
  DETAIL = 'detail',
  OTHER = 'other'
}

export interface ISearchHistoryItem {
  keyword: string;
  lastUpdatedAt: DateLike;
}

export enum TooltipKey {
  EARN = 'earn',
  REWARDS = 'rewards',
  MISSIONS = 'missions',
  STREAK = 'streak'
}

export interface IApiOverrideRule {
  key: string;
  description?: string;
  matcher: string; // reg exp
  status: number;
  response: any;
}

export namespace RemoteConfigModel {
  export interface IAppVersion {
    suggested: number;
    required: number;
  }

  export interface IEarnSection {
    name: string;
    order: number;
    visible: boolean;
    title: string;
    description: string;
    seeAllButton: boolean;
  }
}

export enum RUNTIME_ENV {
  LIVE = 'live',
  DEV_TEST = 'test'
}

export interface IMeasure {
  x: number;
  y: number;
  width: number;
  height: number;
}

export interface IExpirePointsBannerDataSet {
  lastPointsDate: DateLike;
  lastAcceptedDate: DateLike;
}

export interface ISailthruMessage {
  createdAt: string;
  custom: {
    [k: string]: string;
  };
  htmlText: string;
  id: string;
  isRead: boolean;
  text: string;
  title: string;
  type: string;
  cardImageUrl?: string;
  share: boolean;
}

export interface ITutorialStatus {
  isBannerWatched: boolean;
  date: string;
}
