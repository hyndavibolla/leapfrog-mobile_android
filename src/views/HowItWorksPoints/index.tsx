export { default as HowItWorksPoints } from './HowItWorksPoints';
