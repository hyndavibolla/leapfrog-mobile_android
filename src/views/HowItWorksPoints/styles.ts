import { css } from '@emotion/native';

import { COLOR, FONT_FAMILY, FONT_SIZE } from '../../constants';

export const styles = {
  container: css`
    padding: 0 40px 52px;
    background-color: ${COLOR.LIGHT_GRAY};
  `,
  title: css`
    color: ${COLOR.BLACK};
    font-size: ${FONT_SIZE.MEDIUM};
    text-align: center;
    margin-top: 32px;
  `,
  footerTextEmphasis: css`
    color: ${COLOR.PRIMARY_BLUE};
  `,
  section: css`
    margin-top: 32px;
  `,
  subtitle: css`
    font-family: ${FONT_FAMILY.BOLD};
    font-size: ${FONT_SIZE.SMALLER};
    color: ${COLOR.BLACK};
    line-height: ${FONT_SIZE.REGULAR};
    margin-top: 16px;
  `,
  paragraph: css`
    margin-top: 8px;
    font-family: ${FONT_FAMILY.MEDIUM};
    color: ${COLOR.DARK_GRAY};
    font-size: ${FONT_SIZE.SMALLER};
    line-height: ${FONT_SIZE.BIG};
  `,
  arrowContainer: css`
    background-color: ${COLOR.ORANGE};
    width: 18px;
    height: 18px;
    border-radius: 18px;
    justify-content: center;
    align-items: center;
  `,
  lockContainer: css`
    opacity: 0.8;
  `
};
