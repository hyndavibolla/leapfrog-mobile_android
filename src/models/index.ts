import * as GeneralModel from './general';
import * as UserModel from './user';
import * as ActivityModel from './activity';
import * as OfferModel from './offer';
import * as GameModel from './game';
import * as MissionModel from './mission';
import * as RewardModel from './reward';
import * as StreakModel from './streak';
import * as PromptModel from './prompt';
import * as CardLink from './cardLink';

export { GeneralModel, UserModel, ActivityModel, OfferModel, GameModel, MissionModel, RewardModel, StreakModel, PromptModel, CardLink };
