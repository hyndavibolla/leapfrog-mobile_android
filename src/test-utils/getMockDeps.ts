import { ENV } from '_constants';
import { GeneralModel, MissionModel } from '_models';
import { RUNTIME_ENV } from '_models/general';
import { ICrashlyticsService } from '_services/CrashlyticsService';
import { IHttpService } from '_services/HttpService';
import { IPerformanceTrackingService } from '_services/PerformanceTrackingService';
import { getInitialState } from '_state_mgmt/GlobalState';
import {
  getUser,
  getGame,
  getActivity_1,
  getActivity_2,
  getActivity_3,
  getMissionCategory_1,
  getMissionCategory_2,
  getMissionCategory_3,
  getMissionKeyword_1,
  getMissionKeyword_2,
  getMissionKeyword_3,
  getMission_1,
  getRewardConfig_1,
  getSlideObject_1,
  getSlideObject_2,
  getSlideCategoryIdList,
  getStreak_1,
  getQuestion_1,
  getQuestion_2,
  getQuestion_3,
  getAnswer_1,
  getAnswer_2,
  getAnswer_3,
  getNewOnMaxMessage_1,
  getNewOnMaxMessage_2,
  getNewOnMaxMessage_3,
  getSailthruMessage_1,
  getLinkedCards_1,
  getLinkedCards_2,
  getLinkedCards_3,
  getLinkedCards_4,
  getNewOnMaxMessage_4,
  getGiftCard,
  getGiftCardDetail,
  getGiftCardBalance
} from '_test_utils/entities';
import { createUUID } from '_utils/create-uuid';

export const getMockDeps = (initialStateGetter = getInitialState): GeneralModel.Deps => ({
  apiService: {
    isTokenInvalid: false,
    getDeviceIdAsync: async () => createUUID(),
    isTokenSetCurrentlyReady: () => true,
    setTokenSet: jest.fn(),
    authenticate: jest
      .fn()
      .mockResolvedValue({ accessToken: 'accessToken', refreshToken: 'refreshToken', accessTokenExpiryTime: 'accessTokenExpiryTime', deviceId: 'deviceId' }),
    enrollCard: jest.fn().mockResolvedValue(undefined),
    fetchRefId: jest.fn().mockResolvedValue('test-ref-id'),
    fetchUserProfile: jest.fn().mockResolvedValue(getUser()),
    fetchGameState: jest.fn().mockResolvedValue(getGame()),
    fetchRewardConfig: jest.fn().mockResolvedValue(getRewardConfig_1()),
    fetchActivityHistory: jest.fn().mockResolvedValue([getActivity_1(), getActivity_2(), getActivity_3()]),
    fetchGiftCardList: jest.fn().mockResolvedValue([getGiftCard()]),
    fetchGiftCardDetail: jest.fn().mockResolvedValue(getGiftCardDetail()),
    fetchGiftCardBalance: jest.fn().mockResolvedValue(getGiftCardBalance()),
    fetchWalletStats: jest.fn().mockResolvedValue({
      totalLifetimeEarnedPoints: 15550,
      monthlySummaries: {
        '2020-02': 1500,
        '2020-03': 2000,
        '2020-04': 100,
        '2020-05': 950,
        '2020-06': 1000,
        '2020-07': 1500
      }
    }),
    fetchMissionCategoryList: jest.fn().mockResolvedValue([getMissionCategory_1(), getMissionCategory_2(), getMissionCategory_3()]),
    fetchMissionKeywordList: jest.fn().mockResolvedValue({
      [MissionModel.KeywordType.CATEGORY]: [getMissionKeyword_1().value],
      [MissionModel.KeywordType.BRAND]: [getMissionKeyword_2().value, getMissionKeyword_3().value]
    }),
    fetchMissionList: jest.fn(async ({ listType }) => ({ userId: 'buttonId', missions: [getMission_1()], title: 'mission endpoint title', listType })),
    registerDeviceInfo: jest.fn(),
    registerUserActivity: jest.fn(),
    logout: jest.fn(),
    getSYWOnlineToken: jest.fn().mockResolvedValue('test-token'),
    sendValidationEmail: jest.fn().mockResolvedValue(undefined),
    fetchStreakList: jest.fn().mockResolvedValue([getStreak_1()]),
    fetchPromptList: jest.fn().mockResolvedValue({
      promptQuestions: [getQuestion_1(), getQuestion_2(), getQuestion_3()],
      answeredQuestions: [getAnswer_1(), getAnswer_2(), getAnswer_3()]
    }),
    updatePromptList: jest.fn(),
    unregisterDeviceInfo: jest.fn(),
    fetchLinkedCardsList: jest.fn().mockResolvedValue({ linkedCards: [getLinkedCards_1(), getLinkedCards_2(), getLinkedCards_3(), getLinkedCards_4()] }),
    activateLocalOffer: jest.fn(),
    fetchAppleWalletPass: jest.fn().mockResolvedValue('test-pass'),
    updateCardStatus: jest.fn()
  } as any,
  appService: {
    getDeviceIdAsync: async () => createUUID()
  } as any,
  crashlyticsService: {
    initialize: jest.fn(),
    log: jest.fn(),
    recordError: jest.fn(),
    setCrashlyticsCollectionEnabled: jest.fn()
  } as ICrashlyticsService,
  httpService: {
    fetch: jest.fn()
  } as IHttpService,
  performanceTrackingService: {
    initialize: jest.fn(),
    measuredFetch: jest.fn(),
    setPerformanceCollectionEnabled: jest.fn()
  } as IPerformanceTrackingService,
  raiseService: {
    fetchSlideBrand: jest.fn().mockResolvedValue([getSlideObject_1(), getSlideObject_2()]),
    fetchSlideCategoryIdList: jest.fn().mockResolvedValue(getSlideCategoryIdList())
  } as any,
  nativeHelperService: {
    alert: {
      alert: jest.fn(),
      prompt: jest.fn()
    },
    appState: { addEventListener: jest.fn(), currentState: 'active' },
    storage: {
      get: jest.fn().mockResolvedValue(null),
      set: jest.fn().mockResolvedValue(null),
      remove: jest.fn().mockResolvedValue(null)
    },
    clipboard: {
      getString: jest.fn(),
      setString: jest.fn()
    },
    dimensions: {
      getWindowWidth: jest.fn().mockReturnValue(750),
      getWindowHeight: jest.fn().mockReturnValue(1000),
      isSmallDevice: false
    },
    deviceInfo: {
      getDeviceId: createUUID,
      getUniqueId: createUUID,
      getVersion: () => '0.0',
      getBuildNumber: () => '0.1',
      getDeviceModel: () => 'iPhone 11',
      getLocales: { languageTag: 'en-US' },
      getBaseOs: async () => 'iOS',
      getSystemVersion: () => '11.0',
      getDeviceToken: () => createUUID(),
      getDeviceTokenAsync: async () => createUUID(),
      getDeviceOsType: async () => 'iOS',
      getAppVersion: () => '0.0.0.1',
      getDeviceLanguage: () => 'en-US',
      getDeviceOsVersion: () => '11.0',
      getFCMTokenAsync: async () => createUUID(),
      getTimeZone: () => 'testing-timezone',
      getReadableVersion: () => '1.1.1.1',
      onFCMTokenRefresh: jest.fn()
    },
    netInfo: {
      getState: jest.fn().mockResolvedValue({ isConnected: true }),
      subscribeToConnectionChanges: jest.fn(),
      waitForConnection: jest.fn().mockResolvedValue(undefined)
    },
    linking: {
      openURL: jest.fn(() => Promise.resolve()),
      canOpenURL: jest.fn(),
      checkURLScheme: jest.fn().mockResolvedValue(true),
      getInitialURL: jest.fn(),
      addEventListener: jest.fn()
    },
    buttonSdk: {
      configure: jest.fn(),
      setUserId: jest.fn(),
      clearData: jest.fn(),
      purchaseRequest: jest.fn()
    },
    platform: {
      os: 'default',
      select: jest.fn(config => config?.default)
    },
    fireBaseMessaging: {
      getToken: jest.fn().mockReturnValue(createUUID())
    },
    hash: {
      buildSHA256: jest.fn().mockResolvedValue('hash')
    },
    restart: jest.fn(),
    cookieManager: {
      clearAll: jest.fn()
    },
    base64: {
      encode: jest.fn((...args) => `b64 encoded: ${JSON.stringify(args)}`),
      decode: jest.fn((...args) => `b64 decoded: ${JSON.stringify(args)}`)
    },
    getRuntimeEnv: jest.fn().mockResolvedValue(RUNTIME_ENV.DEV_TEST),
    messaging: {
      hasPermission: jest.fn().mockReturnValue(true),
      requestPermission: jest.fn()
    },
    pushNotificationsAuthStatus: {
      NOT_DETERMINED: -1,
      DENIED: 0,
      AUTHORIZED: 1,
      PROVISIONAL: 2
    },
    sailthru: {
      getMessages: jest
        .fn()
        .mockResolvedValue([getNewOnMaxMessage_1(), getNewOnMaxMessage_4(), getNewOnMaxMessage_2(), getNewOnMaxMessage_3(), getSailthruMessage_1()]),
      markMessageAsRead: jest.fn().mockResolvedValue(true),
      getInitialNotification: jest.fn(),
      onNotificationTapped: jest.fn(cb => {
        cb({});
        return () => {};
      }) as any
    },
    reactNativePermission: require('react-native-permissions/mock'),
    geolocation: {
      addListener: jest.fn(),
      getCurrentPosition: jest.fn(),
      removeListeners: jest.fn(),
      requestAuthorization: jest.fn(),
      setConfiguration: jest.fn(),
      startObserving: jest.fn(),
      stopObserving: jest.fn()
    }
  } as any,
  stateSnapshot: {
    set: jest.fn(),
    get: jest.fn().mockReturnValue(initialStateGetter())
  },
  logger: {
    log: jest.fn(),
    info: jest.fn(),
    debug: jest.fn(),
    warn: jest.fn(),
    error: jest.fn(),
    success: jest.fn(),
    color: jest.fn(),
    assert: jest.fn(),
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    format: jest.fn()
  } as any,
  eventTrackerService: {
    trackDataUser: jest.fn(),
    isTrackingEnabled: jest.fn().mockResolvedValue(true),
    rZero: {
      trackView: jest.fn(),
      trackEvent: jest.fn(),
      flush: jest.fn(),
      setUser: jest.fn()
    },
    forterSDK: {
      setDevLogsEnabled: jest.fn(),
      getDeviceUniqueID: jest.fn(cb => cb()),
      init: jest.fn((a, b, c) => c()),
      trackNavigation: jest.fn(),
      trackNavigationWithExtraData: jest.fn(),
      trackAction: jest.fn(),
      trackActionWithJSON: jest.fn(),
      fraudTrackNavigationEvent: jest.fn(),
      fraudTrackAction: jest.fn()
    },
    tealiumSDK: {
      track: jest.fn(),
      trackView: jest.fn()
    },
    appsFlyerSDK: {
      updateServerUninstallToken: jest.fn(),
      setCustomerUserId: jest.fn().mockImplementation((_, cb) => cb('success')),
      addDeepLinkEventListener: jest.fn(),
      removeDeepLinkEventListener: jest.fn(),
      clearDeepLinkEventListener: jest.fn(),
      trackConversionEvent: jest.fn()
    }
  } as any,
  ccpaService: {
    showConsentBanner: jest.fn().mockResolvedValue(null),
    showPreferenceCenter: jest.fn()
  } as any,
  remoteConfigService: {
    getValue: jest.fn(async key => ENV.REMOTE_CONFIG.DEFAULT_CONFIG[key]),
    fetchAndActivate: jest.fn().mockResolvedValue(null)
  } as any,
  authService: {
    auth: {
      refreshToken: jest.fn().mockResolvedValue(true)
    },
    webAuth: {
      authorize: jest.fn(),
      clearSession: jest.fn()
    }
  } as any
});
