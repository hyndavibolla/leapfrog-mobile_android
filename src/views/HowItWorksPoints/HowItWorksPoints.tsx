import React, { memo } from 'react';
import { View, ScrollView } from 'react-native';

import { Text } from '_components/Text';
import { Icon } from '_commons/components/atoms/Icon';
import { COLOR, FONT_FAMILY, FONT_SIZE, ICON } from '_constants';

import { styles } from './styles';

export const HowItWorksPoints = () => {
  return (
    <ScrollView>
      <View style={styles.container}>
        <Text font={FONT_FAMILY.HEAVY} style={styles.title}>
          How do points work?
        </Text>
        <View style={styles.section}>
          <Icon name={ICON.SYW_CIRCLE} size={FONT_SIZE.SMALLER} />
          <Text style={styles.subtitle}>Available points:</Text>
          <Text style={styles.paragraph}>
            Available points are ready to be redeemed! Use points for Gift Cards in the Rewards section of MAX, or on our Marketplace partner products. Treat
            yourself! You’ve earned it!
          </Text>
        </View>
        <View style={styles.section}>
          <View style={styles.lockContainer}>
            <Icon name={ICON.LOCK_CIRCLE} color={COLOR.DARK_GRAY} size={FONT_SIZE.SMALL} />
          </View>
          <Text style={styles.subtitle}>Pending points:</Text>
          <Text style={styles.paragraph}>
            After you make a qualifying purchase with a merchant, the points you’ve earned will be in a “pending” state until the purchase has been cleared with
            the merchant. Please allow up to ninety days to receive the points back for any mobile or online purchases.
          </Text>
        </View>
        <View style={styles.section}>
          <View style={styles.arrowContainer}>
            <Icon name={ICON.ARROW_DOWN} color={COLOR.WHITE} size={FONT_SIZE.TINY} />
          </View>
          <Text style={styles.subtitle}>Expiring points:</Text>
          <Text style={styles.paragraph}>
            All points earned have an expiration date. We’ll be sure to let you know when they are getting close! After points are expired, those points will
            not be eligible for redemption and removed from your available point balance.
          </Text>
        </View>
      </View>
    </ScrollView>
  );
};

export default memo(HowItWorksPoints);
