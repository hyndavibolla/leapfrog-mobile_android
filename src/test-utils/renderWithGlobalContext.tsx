import React, { ReactElement } from 'react';
import { render } from '@testing-library/react-native';

import { getInitialState, GlobalProvider } from '../state-mgmt/GlobalState';
import { Deps, IGlobalState, IReducer } from '../models/general';
import { getMockDeps } from './getMockDeps';
import { getMockReducer } from './getMockReducer';

export const renderWithGlobalContext = (
  children: ReactElement,
  deps: Deps = getMockDeps(),
  initState: IGlobalState = getInitialState(),
  combinedReducers: IReducer<IGlobalState, any> = getMockReducer()
) => {
  const renderAPI = render(
    <GlobalProvider deps={deps} initState={initState} combinedReducers={combinedReducers}>
      {children}
    </GlobalProvider>
  );
  return { ...renderAPI, mockReducer: combinedReducers, deps };
};
