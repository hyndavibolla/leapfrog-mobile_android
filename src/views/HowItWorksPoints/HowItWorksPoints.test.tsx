import React from 'react';

import HowItWorksPoints from './HowItWorksPoints';
import { renderWithGlobalContext } from '../../test-utils/renderWithGlobalContext';
import { Deps } from '../../models/general';
import { getMockDeps } from '../../test-utils/getMockDeps';

describe('HowItWorksPoints', () => {
  let deps: Deps;

  beforeEach(() => {
    deps = getMockDeps();
  });

  it('should render', () => {
    const { toJSON } = renderWithGlobalContext(<HowItWorksPoints />, deps);
    expect(toJSON()).toMatchSnapshot();
  });
});
