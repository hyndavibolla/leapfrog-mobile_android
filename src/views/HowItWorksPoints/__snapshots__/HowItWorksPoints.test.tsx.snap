// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HowItWorksPoints should render 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "#F5F7FA",
          "paddingBottom": 52,
          "paddingLeft": 40,
          "paddingRight": 40,
          "paddingTop": 0,
        }
      }
    >
      <Text
        font="SFProDisplay-Heavy"
        style={
          Array [
            Object {
              "color": "#909BA0",
              "fontFamily": "SFProDisplay-Heavy",
              "fontSize": 16,
              "lineHeight": 22,
            },
            Object {
              "color": "#262C2F",
              "fontSize": 22,
              "marginTop": 32,
              "textAlign": "center",
            },
          ]
        }
      >
        How do points work?
      </Text>
      <View
        style={
          Object {
            "marginTop": 32,
          }
        }
      >
        <Text
          accessibilityLabel="icon-text"
          accessible={true}
          allowFontScaling={false}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#0066CC",
                "fontSize": 16,
              },
              undefined,
              Object {
                "fontFamily": "sywmax",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="icon-text"
        >
          
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#909BA0",
                "fontFamily": "SFProDisplay-Medium",
                "fontSize": 16,
                "lineHeight": 22,
              },
              Object {
                "color": "#262C2F",
                "fontFamily": "SFProDisplay-Bold",
                "fontSize": 16,
                "lineHeight": 20,
                "marginTop": 16,
              },
            ]
          }
        >
          Available points:
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#909BA0",
                "fontFamily": "SFProDisplay-Medium",
                "fontSize": 16,
                "lineHeight": 22,
              },
              Object {
                "color": "#909BA0",
                "fontFamily": "SFProDisplay-Medium",
                "fontSize": 16,
                "lineHeight": 24,
                "marginTop": 8,
              },
            ]
          }
        >
          Available points are ready to be redeemed! Use points for Gift Cards in the Rewards section of MAX, or on our Marketplace partner products. Treat yourself! You’ve earned it!
        </Text>
      </View>
      <View
        style={
          Object {
            "marginTop": 32,
          }
        }
      >
        <View
          style={
            Object {
              "opacity": 0.8,
            }
          }
        >
          <Text
            accessibilityLabel="icon-text"
            accessible={true}
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#909BA0",
                  "fontSize": 18,
                },
                undefined,
                Object {
                  "fontFamily": "sywmax",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
            testID="icon-text"
          >
            
          </Text>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#909BA0",
                "fontFamily": "SFProDisplay-Medium",
                "fontSize": 16,
                "lineHeight": 22,
              },
              Object {
                "color": "#262C2F",
                "fontFamily": "SFProDisplay-Bold",
                "fontSize": 16,
                "lineHeight": 20,
                "marginTop": 16,
              },
            ]
          }
        >
          Pending points:
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#909BA0",
                "fontFamily": "SFProDisplay-Medium",
                "fontSize": 16,
                "lineHeight": 22,
              },
              Object {
                "color": "#909BA0",
                "fontFamily": "SFProDisplay-Medium",
                "fontSize": 16,
                "lineHeight": 24,
                "marginTop": 8,
              },
            ]
          }
        >
          After you make a qualifying purchase with a merchant, the points you’ve earned will be in a “pending” state until the purchase has been cleared with the merchant. Please allow up to ninety days to receive the points back for any mobile or online purchases.
        </Text>
      </View>
      <View
        style={
          Object {
            "marginTop": 32,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FF5F00",
              "borderBottomLeftRadius": 18,
              "borderBottomRightRadius": 18,
              "borderTopLeftRadius": 18,
              "borderTopRightRadius": 18,
              "height": 18,
              "justifyContent": "center",
              "width": 18,
            }
          }
        >
          <Text
            accessibilityLabel="icon-text"
            accessible={true}
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 12,
                },
                undefined,
                Object {
                  "fontFamily": "sywmax",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
            testID="icon-text"
          >
            
          </Text>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#909BA0",
                "fontFamily": "SFProDisplay-Medium",
                "fontSize": 16,
                "lineHeight": 22,
              },
              Object {
                "color": "#262C2F",
                "fontFamily": "SFProDisplay-Bold",
                "fontSize": 16,
                "lineHeight": 20,
                "marginTop": 16,
              },
            ]
          }
        >
          Expiring points:
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#909BA0",
                "fontFamily": "SFProDisplay-Medium",
                "fontSize": 16,
                "lineHeight": 22,
              },
              Object {
                "color": "#909BA0",
                "fontFamily": "SFProDisplay-Medium",
                "fontSize": 16,
                "lineHeight": 24,
                "marginTop": 8,
              },
            ]
          }
        >
          All points earned have an expiration date. We’ll be sure to let you know when they are getting close! After points are expired, those points will not be eligible for redemption and removed from your available point balance.
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
